src/index/base.cpp

```diff
@@ -370,3 +366,11 @@ IndexSummary BaseIndex::GetSummary() const
     summary.best_block_height = m_best_block_index ? m_best_block_index.load()->nHeight : 0;
     return summary;
 }
+
+void BaseIndex::SetBestBlockIndex(const CBlockIndex* block) {
+    m_best_block_index = block;
+    if (AllowPrune() && (fPruneMode || fHavePruned) && !fReindex) {
+        LOCK(::cs_main);
+        g_chainman.m_blockman.UpdatePruneBlocker(GetName(), block ? block : ::ChainActive().Genesis());
```

Comment:
Error building this branch at e365e87a72cc4 rebased to current master at 8cdf91735f2b, looks like it needs updating:
```rake
index/base.cpp:373:9: error: use of undeclared identifier 'g_chainman'
        g_chainman.m_blockman.UpdatePruneBlocker(GetName(), block ? block : ::ChainActive().Genesis());
        ^
index/base.cpp:373:79: error: no member named 'ChainActive' in the global namespace
        g_chainman.m_blockman.UpdatePruneBlocker(GetName(), block ? block : ::ChainActive().Genesis());
                                                                            ~~^
2 errors generated.
```

Reply:
Thanks, fixed.